components/dashboard/src/user-settings/Preferences.tsx

```diff
@@ -82,34 +83,36 @@ export default function Preferences() {
 
                 <Heading2 className="mt-12">Dotfiles</Heading2>
                 <Subheading>Customize workspaces using dotfiles.</Subheading>
+
                 <form className="mt-4 max-w-xl" onSubmit={saveDotfileRepo}>
-                    <h4>Repository URL</h4>
-                    <span className="flex">
-                        <input
-                            type="text"
-                            value={dotfileRepo}
-                            className="w-96 h-9"
-                            placeholder="e.g. https://github.com/username/dotfiles"
-                            onChange={(e) => setDotfileRepo(e.target.value)}
-                        />
-                        <Button type="secondary" className="ml-2">
-                            Save Changes
-                        </Button>
-                    </span>
-                    <div className="mt-1">
-                        <p className="text-gray-500 dark:text-gray-400">
-                            Add a repository URL that includes dotfiles. Gitpod will
-                            <br />
-                            clone and install your dotfiles for every new workspace.
-                        </p>
-                    </div>
+                    <InputField
+                        label="Repository URL"
+                        hint="Add a repository URL that includes dotfiles. Gitpod will clone and install your dotfiles for every new workspace."
```

Comment:
Good eye 🦅 👁️  - it was a side effect of re-using the form components here, and finding a way to add the button into the row the input was on, but keep the `hint` rendering as it did by default. I'll play around with this a little more to see if there's a better way to have that behavior, but still re-use the form input components.

Reply:
Ah, I see. Yeah, let's make this change if it's trivial—otherwise we could keep it like this, as I wouldn't be surprized if the UX or interaction to update the dotfiles or timeouts moves away from these inline forms, see relevant comments:
- https://github.com/gitpod-io/gitpod/issues/7385#issuecomment-1278860196
- https://github.com/gitpod-io/gitpod/issues/7385#issuecomment-1372436421